pallets/ddc-customers/src/migration.rs

```diff
@@ -78,10 +106,50 @@ pub fn migrate_to_v1<T: Config>() -> Weight {
 	}
 }
 
+// Migrate to removable buckets
+pub fn migrate_from_v1_to_v2<T: Config>() -> Weight {
+	let on_chain_version = Pallet::<T>::on_chain_storage_version();
+	if on_chain_version == 1 {
+		let count = v1::BucketsCount::<T>::get();
+		info!(
+			target: LOG_TARGET,
+			" >>> Updating DDC Customers storage. Migrating {} buckets...", count
+		);
+
+		Buckets::<T>::translate::<v1::Bucket<T::AccountId>, _>(
+			|bucket_id: BucketId, bucket: v1::Bucket<T::AccountId>| {
+				info!(target: LOG_TARGET, "     Migrating bucket for bucket ID {:?}...", bucket_id);
+
+				Some(Bucket {
+					bucket_id: bucket.bucket_id,
+					owner_id: bucket.owner_id,
+					cluster_id: bucket.cluster_id,
+					is_public: bucket.is_public,
+					is_removed: false,
+					total_customers_usage: None,
+				})
+			},
+		);
+
+		// Update storage version.
+		StorageVersion::new(2).put::<Pallet<T>>();
+		let count = v1::BucketsCount::<T>::get();
+		info!(
+			target: LOG_TARGET,
+			" <<< DDC Customers storage updated! Migrated {} buckets ✅", count
+		);
+
+		T::DbWeight::get().reads_writes(count + 2, count + 1)
+	} else {
+		info!(target: LOG_TARGET, " >>> Unused migration!");
+		T::DbWeight::get().reads(1)
+	}
+}
+
 pub struct MigrateToV1<T>(sp_std::marker::PhantomData<T>);
 impl<T: Config> OnRuntimeUpgrade for MigrateToV1<T> {
 	fn on_runtime_upgrade() -> Weight {
-		migrate_to_v1::<T>()
+		migrate_from_v1_to_v2::<T>()
```

Comment:
You should provide a new `MigrateToV2<T>` struct and call the `migrate_from_v1_to_v2` from there, instead of modifying the old `MigrateToV1<T>`.

Reply:
And after that, add the `MigrateToV2<T>` to the type `Migrations = (...)` tuple in the runtime lib file

Reply:
Agreed. @aie0 Could you please address above comments related to migration?